### DISCUSSION TITLE
Discussion - Plex-Meta-Manager

### DISCUSSION THREAD

**Discussion by j-ktz** [CodyBerenson/PGMA-Modernized#74] [General]
Hey Boys!

I've recently found and fell in love with [Plex-Meta-Manager.](https://github.com/meisnate12/Plex-Meta-Manager) I have it running in Docker every night at 3AM and makes collections based on a Movies.yml code (auto generated Genre, Studio, Trakt, IMDB, etc. Collections). It's GREAT. So it got be thinking of I could retrofit for my XXX libraries, and I can!

I did some basic studio collections and it worked. For example, I can have it run at 3AM with my Movies.yml code and it'll find any scene/movie that's has studio = seancody, sean cody, seancody.com and add it to the Sean Cody collection. This saves me so much time for manually tagging scenes, which is what I was doing before. It could work for Genres, Stars, etc. You can even add in Summaries and Poster art for Collections. What Jason is doing with IAFD Collections is already phenomenal but this could be a great addition too!

Just getting started but happy to share my code and posters for studios and let you guys know how to modify for your own collections. Interested? 

Also @CodyBerenson is it possible to have a Discussions tab for the Repo? This isn't really an issue but didn't know where else to post. :)

**Comment by CodyBerenson** [comment]
@j-ktz Discussions now enabled.  Thanks for the suggestion.

**Comment by joca879** [comment]
Hi @j-ktz 
I am a novice in terms of plex, I am now taking the first steps and building my libraries, regular and XXX.
@CodyBerenson and @JPH71 have given me a great help in achieving this.
This automation seems interesting to me in order to create automatic collections, not only in terms of XXX content (the features that JPH1 have been developing lately already go in that direction) but also for general films and series. 
I am still a bit late compared to you, as far as the size of your XXX libraries is concerned, but could you tell me how you implemented this feature, how you changed it for the XXX collections and whether to show or share your posters.
Cheers!

**Reply by j-ktz** [reply]
Welcome to the party @joca879! Yup, so linked above, I've used Plex Meta Manager. It's pretty easy to setup, and the crew over there is super helpful. Essentially you have a config file (that has all your library and Plex info), and then a YML file for each library (Movies, TV Shows, XXX). In those files is where you set up your collection info. I've attached my XXX one for reference so you can see how I set mine up. I have mine set up with Collection name, description, sort order and then I manually added my posters. However, you can link to external posters too! There's examples of that over on the Plex Meta Manager GitHub. 

[XXX - Studio Porn.yml.zip](https://github.com/CodyBerenson/PGMA-Modernized/files/6126085/XXX.-.Studio.Porn.yml.zip)

**Reply by joca879** [reply]
Thanks @j-ktz!
I´ve just downloaded and see your file... you have a good and huge collection... very nice ;)
I'll try and see if I can make it work for me

**Reply by j-ktz** [reply]
@joca879 Haha, I'm a hoarder. I've also uploaded my posters to Google Drive if you want to use any! https://drive.google.com/drive/folders/1nzB7SW3yxSJVcA-S6FCFXjwZutqE8XN6?usp=sharing

**Reply by joca879** [reply]
Thanks a lot @j-ktz ... That's phenomenal, you have a lot of work there
Thanks for sharing :)

**Reply by joca879** [reply]
Hey @j-ktz!

You left me with the bug grinding on my head ... I'm a newbie in NAS and plex (I'm running the plex media server in a synology), and in these poster things. Can you tell me how you have created your posters? What application/software do you use? That scene of the text behind the image is interesting... and also the use of studio logos. If you prefer, you can send me an email, no problem: joca8791@hotmail.com

Cheers!

**Reply by j-ktz** [reply]
@joca879 I'm still new to Syno/NAS too but have learned a lot over the past few years! I spent way too much time in Photoshop creating all the posters myself! I wonder if my college professors knew this was how I was going to use my design degree. 😆

**Reply by joca879** [reply]
@j-ktz I was suspicious that it should be Photoshop or some similar application, it has a professional and clean look. At least you're applying what you've learned :) It's an application that I have never used.

**Reply by CodyBerenson** [reply]
@j-ktz if I already asked this, I apologize...not sure it went thru.  What is the font you use in your stars collection.  I'd like to match your work as closely as possible.  :)  thanks!

**Reply by CodyBerenson** [reply]
Franklin Gothic Demi Cond gets me relatively close results......thanks again!

**Reply by j-ktz** [reply]
@CodyBerenson I use Helvetica Neue LT Std, 87 Heavy Condensed but Franklin Gothic is pretty close. 😉 If you struggle with any of it and need help just holler.

**Reply by CodyBerenson** [reply]
Was afraid it was a font I don't have.  LOL.  Franklin will have to do. Its pretty close...

![image](https://user-images.githubusercontent.com/40373801/111001287-80b70600-8351-11eb-9ad4-213c64c890c3.png)

**Reply by CodyBerenson** [reply]
@j-ktz I'm not sure I'm grateful....or furious....that I've started down the collections path.  I'm going to lose the next year of my life on this.  LOL

**Reply by joca879** [reply]
@CodyBerenson I agree with you, the desire to invest in collections emerged as soon as I see the work of @j-ktz

I'm tempted to do the same, although I know I'm going to take a lot of time creating something like that... But first I need to learn to work with photoshop lol

**Reply by j-ktz** [reply]
@CodyBerenson That look great!!! I added the font I used to the Google Drive if you want it exactly the same. Also allowed you all to add to the Drive too if you wanna share the posters you make. :)

**Reply by CodyBerenson** [reply]
THANKS!

**Reply by joshallen13** [reply]
> @j-ktz I'm not sure I'm grateful....or furious....that I've started down the collections path. I'm going to lose the next year of my life on this. LOL

@CodyBerenson This is exactly the reason I wanted to show off that coverlabs.io designer, as you can pretty much drag and drop without spending too much time on this, as I could very easily spend an hour or two in a rabbit hole online trying to figure out how to master an obscure photoshop feature.

@j-ktz Don't worry about being alone in utilizing your degree! I'm sure my art history professors are assuming my degree is being best applied to bullshitting my way through an explanation or extracting symbolism in all this pornz. I mean, not much else I can use it for... :)

**Reply by CodyBerenson** [reply]
So I went into full-on OCD mode, worrying that I'd miss a favorite star as I was building my collections...and the panic level grew.  But then I thought, duh, just query the database and choose from the actors that I want to make collections for.  

In case you want to use SQLite, here's the SQL (I took a copy of the database so there was zero chance I'd mess it up).  For my films library (in my DB it is library_section_id 10 and metadata_type 1), I wanted to know what actors have the most occurrences (because I don't want to make a collection for an actor in only 1 or two movies):

SELECT tags.tag, count(taggings.tag_ID)
FROM taggings 
left JOIN tags on taggings.tag_ID = tags.id
left join metadata_items on taggings.metadata_item_id = metadata_items.id
where metadata_items.library_section_id = 10 
and metadata_items.metadata_type = 1
and tags.tag_type = 6
group by taggings.tag_id
order by count(taggings.tag_ID) DESC

![image](https://user-images.githubusercontent.com/40373801/111074896-a6f8b500-84bb-11eb-9bfd-0e828c0d73e4.png)

**Reply by CodyBerenson** [reply]
@j-ktz I lost a day and a half of my life creating covers for my actors collection.  LOL!  I can't thank you enough for the inspiration.  I'm taking a break for now, and will upload my covers and my template into your drive.  I was on such a roll that after a while, I completely forgot to check to see if you had already created a certain actor or not...hence all the duplicates. 

I only did my films library.  I have my films in a completely separate library from my scenes...the scenes are in a library per studio.  I wish Plex was more elegant in how it handles collections across libraries.  So, I'm still strategizing on how to tackle the scenes collections....maybe i just won't.   I'm definitely open to ideas.

All the best, Cody.

**Reply by j-ktz** [reply]
@CodyBerenson Welcome to the dark side!!! Glad you tackled it. Can't wait to see all the posters you created. You probably know this BUT if you create a Collection in your scenes library with the same name as your films library they'll all lump together. For example, if you have Austin Wilde in films, and you open that Collection, at the bottom of the collection it will see "Movies with Austin Wilde" and then display the scenes tagged with "Austin Wilde" as well. Clicking on the hyperlink will take you to the other Collection in scenes. Then you can just use the same poster. That's basically how I handle porn and fan site content. 

![image](https://user-images.githubusercontent.com/40373801/111306093-2ad2ae80-862e-11eb-818c-93ed5250f9b2.png)

**Reply by CodyBerenson** [reply]
@j-ktz  Yuppp....The challenge is that Cade Maddox, as an example, is in a dozen studio libraries plus the main film library.  it is nice that you can see all of content in one collection...ish....but it is ugly when you search on Cade Maddox and get 13 collections.  lol.  If I do tag each of the scenes libraries, I think I may do the tagging from the backend with SQL inserts....as long as I take plenty of database backups.  LOL.

BUT, VERY interesting....I hadn't even considered managing Only Fan content in Plex.  My head is going to further explode.  STOP sending me down rabbit holes. There goes another two days of my life.   LOL LOL LOL.

**Reply by j-ktz** [reply]
@CodyBerenson Yeah, Collections are tough if you have separate libraries for each Studio. I have three, Studio Porn, Amateur (think not professional studios, PornHub content, etc.) and OnlyFans. Then in Studio Porn, I use collections to break out Studios (instead of having separate Plex libraries). Happy to invite you to my library if you want to see how I organize everything!

**Reply by joca879** [reply]
Hello @j-ktz 
In fact, collections have their importance in the organization of films and scenes. There is some work, with the creation of posters, but it seens more clean. @CodyBerenson once showed me how he organizes his library ... if you don't mind, could you show how you organize yours, give me a sneek peak? I am still at the beginning and I am not sure if I am doing it the right way ... if your organization is better, I still have time to rectify without having a lot of work :)

**Reply by j-ktz** [reply]
@joca879 Sure! It's not perfect but I'll let you take a look around. Shoot me your email. @CodyBerenson Happy to show you too, just send your email.

**Reply by joca879** [reply]
@j-ktz Here goes my email: joca8791@hotmail.com. Thanks :)

**Reply by j-ktz** [reply]
@joca879 Sent!

**Reply by joca879** [reply]
@j-ktz Thanks! I´ve shared mine with you... compared to yours, it's nothing. I will see how you organize things

**Reply by joca879** [reply]
@j-ktz You really must have spent a lot of time to have the library as it is. It's very good. Thanks for letting me have a look, I already have a clearer idea of ​​how I can do mine!

**Reply by CodyBerenson** [reply]
@j-ktz I truthfully am not sure whether I should thank you or be angry with you.  LOL!  My tagging and cataloging actors has highlighted how much of my library was indexed before every meeting Jason and starting down the PGMA path, so much of my library's metadata was built using legacy scanners that sometimes worked and most times did not...most particularly not being populated with actors. as long as it had the pretty film poster, that's all I cared about.  

so, thanks to you, i'm now relabeling and rematching significant swaths of my library.  LOL.  THANKS A WHOLE HELL OF A LOT.  ;)

**Reply by j-ktz** [reply]
@CodyBerenson Proud of you! You're gonna be the Marie Kondo of porn after this. 😂

**Comment by JPH71** [comment]
When selecting an agent under GayAdult, GayFilm or GayScene - click on the
gearbox symbol and from there you should be able to set what collections
you need setting.

Jason x

**Reply by joca879** [reply]
Hey @JPH71!
Thanks for the instructions, I´ll try the options that you indicate :)

**Comment by CodyBerenson** [comment]
Really nicely done.  I am rethinking my not using collections philosophy.

**Reply by j-ktz** [reply]
They're a game changer.

**Reply by joca879** [reply]
Yes, they are really interesting ... and some of the photos give even more enthusiasm hehehe

**Comment by CodyBerenson** [comment]
Hiya @j-ktz 

Perhaps Plex heard you a wee-bit.  They've just introduced "Smart Collections", which allows one to keep collections up to date...."Any future items that meet the filter criteria will be automatically be added and things which no longer meet the criteria will be be automatically removed"

https://support.plex.tv/articles/201273953-collections/

Sounds a little like the metadata manager you were describing.  So, no sooner did I get my collections just where I want them, now I have to make them into Smart Collections.  

At this point I freaking give up.  LOL.

**Reply by j-ktz** [reply]
OMG. When did this happen?! It's great but now I have to rebuild them all lol.

**Reply by CodyBerenson** [reply]
It was just in the most recent Plex release. Plex is insidious like this.  There was a great bolt on tool called Tattuli (sp?) that would tell you who's watching what when (if you've shared your library with lots of folks)....and then Plex duplicated the functionality in their dashboard.  So, no sooner does a great tool like this metadata manager that you found take hold, and they duplicate the functionality and build it in.    Very Microsoft and Oracle-esque.  LOL

The upside is that as Plex changes its data model and structures one doesn't have to worry what will break in any bolt-on tools if one isn't using any.  

The downside is that I can't imagine how much time and energy Jason put into creating all the runtime collections creation functionality that is now dynamically built in to Plex.  :(

**Reply by j-ktz** [reply]
@CodyBerenson I didn't even hear about it! So weird, I don't have the option yet on the latest PMS and Web App. Also, this doesn't allow us to search Summaries/Descriptions, only Titles. But still cool!
